TypeError when creating security group ingress rule via CloudFormation

- **How to reproduce the issue**
Using a CFT like this:
```
{
    "AWSTemplateFormatVersion": "2010-09-09",
    "Description": "AWS CloudFormation Template to create an EC2 instance",
    "Parameters": {
        "SubnetId": {
            "Type": "String",
            "Description": "An internal subnet ID of a subnet to deploy in",
            "AllowedPattern": "^subnet-[a-zA-Z0-9]*"
        },
        "VPCId": {
            "Type": "String",
            "Description": "The VPC ID",
            "AllowedPattern": "^vpc-[a-zA-Z0-9]*"
        },
        "EC2KeyName": {
            "Type": "String",
            "Description": "The key name assigned to the EC2 instance",
            "Default": "My-Key-Pair"
        },
        "InstanceRoleName": {
            "Type": "String",
            "Description": "The name of the IAM role used by the instance",
            "Default": "My-IAM-Role"
        }
    },
    "Resources": {
        "EC2Instance": {
            "Type": "AWS::EC2::Instance",
            "Properties": {
                "IamInstanceProfile": {
                    "Ref": "IAMInstanceProfile"
                },
                "ImageId": "ami-03657b56516ab7912",
                "InstanceType": "t2.micro",
                "KeyName": {
                    "Ref": "EC2KeyName"
                },
                "SecurityGroupIds": [
                    {
                        "Fn::GetAtt": [
                            "SecurityGroup",
                            "GroupId"
                        ]
                    }
                ],
                "SubnetId": {
                    "Ref": "SubnetId"
                },
                "Tags": [
                    {
                        "Key": "Name",
                        "Value": "My-EC2-Instance"
                    }
                ],
                "Tenancy": "default"
            }
        },
        "SecurityGroup": {
            "Type": "AWS::EC2::SecurityGroup",
            "Properties": {
                "GroupDescription": "Default VPC security group for the instance",
                "GroupName": "My-SG",
                "VpcId": {
                    "Ref": "VPCId"
                }
            }
        },
        "SSHIngressRule": {
            "Type": "AWS::EC2::SecurityGroupIngress",
            "Properties": {
                "CidrIp": "10.0.0.0/8",
                "Description": "Allow SSH traffic from 10.0.0.0/8",
                "FromPort": 22,
                "ToPort": 22,
                "GroupId": {
                    "Fn::GetAtt": [
                        "SecurityGroup",
                        "GroupId"
                    ]
                },
                "IpProtocol": "tcp"
            }
        },
        "IAMInstanceProfile": {
            "Type": "AWS::IAM::InstanceProfile",
            "Properties": {
                "InstanceProfileName": "My-Instance-Profile",
                "Roles": [
                    {
                        "Ref": "InstanceRoleName"
                    }
                ]
            }
        }
    }
}
```
If I have this snippet in a `create_stack.py` file and run this in main (where TEMPLATE is the above JSON file that I've run a `json.load` and `json.dumps` on):
```
import boto3

def create_my_stack(subnet_id, vpc_id):
    cf_client.create_stack(
        StackName=self.stack_name,
        TemplateBody=TEMPLATE,
        Parameters=[
            {
                "ParameterKey": "SubnetId",
                "ParameterValue": subnet_id
            },
            {
                "ParameterKey": "VPCId",
                "ParameterValue": vpc_id
            }
        ],
        Capabilities=['CAPABILITY_NAMED_IAM'],
        OnFailure='DELETE'
    )
```
I have no issues. However if I try to use moto and pytest to assert that I get a response back, I get the following error:
```
self = <moto.ec2.models.EC2Backend object at 0x109fe8890>, group_name_or_id = 'sg-7463a68f', ip_protocol = 'tcp'
from_port = 22, to_port = 22, ip_ranges = ['10.0.0.0/8'], source_group_names = None, source_group_ids = None
vpc_id = None

    def authorize_security_group_ingress(
        self,
        group_name_or_id,
        ip_protocol,
        from_port,
        to_port,
        ip_ranges,
        source_group_names=None,
        source_group_ids=None,
        vpc_id=None,
    ):
        group = self.get_security_group_by_name_or_id(group_name_or_id, vpc_id)
        if ip_ranges:
            if isinstance(ip_ranges, str) or (
                six.PY2 and isinstance(ip_ranges, unicode)  # noqa
            ):
                ip_ranges = [{"CidrIp": str(ip_ranges)}]
            elif not isinstance(ip_ranges, list):
                ip_ranges = [json.loads(ip_ranges)]
        if ip_ranges:
            for cidr in ip_ranges:
>               if not is_valid_cidr(cidr["CidrIp"]):
E               TypeError: string indices must be integers
```
   - **What you expected to happen**
I expected there to be a response, similar to what I get when I actually run the method outside of pytest.

   - **What actually happens**
I get the TypeError from above. It looks like it's looking for a `CidrIp` field in a [dict](https://github.com/spulec/moto/blob/master/moto/ec2/models.py#L2234), rather than the actual CIDR value itself from a list (which is what's being provided). 

   - **What version of Moto you're using** and **How you installed it**
moto==1.3.16
boto3==1.16.34
botocore==1.19.34

I am using the `mock_cloudformation` and `mock_ec2` (to create my fake VPC and subnet to deploy in) and setting them as a pytest fixture to pass into my test case.
In my `test_create_stack.py`
```
def test_create_stack_success(cloudformation, ec2):
    subnet = ec2.describe_subnets()['Subnets'][0]
    subnet_id = subnet['SubnetId']
    vpc_id = subnet['VpcId']
    assert "StackId" in create_stack(subnet_id, vpc_id)
```

In my `conftest.py`:
```
...
@pytest.fixture
def ec2():
    """
    Pytest fixture that creates the test VPC and subnet in the
    fake moto AWS account

    Yields a fake boto3 ec2 client
    """
    with mock_ec2():
        ec2 = boto3.client("ec2", region_name="us-west-2")
        vpc = ec2.create_vpc(
            CidrBlock="172.16.0.0/16",
            AmazonProvidedIpv6CidrBlock=False,
            InstanceTenancy="default"
        )
        vpc_id = vpc['Vpc']['VpcId']
        ec2.create_subnet(
            CidrBlock="172.16.0.0/17",
            VpcId=vpc_id,

        )
        yield ec2
...
```

--- a/moto/ec2/models.py
+++ b/moto/ec2/models.py
@@ -2231,7 +2231,9 @@ class SecurityGroupBackend(object):
                 ip_ranges = [json.loads(ip_ranges)]
         if ip_ranges:
             for cidr in ip_ranges:
-                if not is_valid_cidr(cidr["CidrIp"]):
+                if (type(cidr) is dict and not is_valid_cidr(cidr["CidrIp"])) or (
+                    type(cidr) is str and not is_valid_cidr(cidr)
+                ):
                     raise InvalidCIDRSubnetError(cidr=cidr)
 
         self._verify_group_will_respect_rule_count_limit(
@@ -2432,6 +2434,7 @@ class SecurityGroupIngress(CloudFormationModel):
         group_id = properties.get("GroupId")
         ip_protocol = properties.get("IpProtocol")
         cidr_ip = properties.get("CidrIp")
+        cidr_desc = properties.get("Description")
         cidr_ipv6 = properties.get("CidrIpv6")
         from_port = properties.get("FromPort")
         source_security_group_id = properties.get("SourceSecurityGroupId")
@@ -2458,7 +2461,7 @@ class SecurityGroupIngress(CloudFormationModel):
         else:
             source_security_group_names = None
         if cidr_ip:
-            ip_ranges = [cidr_ip]
+            ip_ranges = [{"CidrIp": cidr_ip, "Description": cidr_desc}]
         else:
             ip_ranges = []
 

--- a/tests/test_ec2/test_security_groups.py
+++ b/tests/test_ec2/test_security_groups.py
@@ -714,15 +714,10 @@ def test_description_in_ip_permissions():
     )
 
     result = conn.describe_security_groups(GroupIds=[sg["GroupId"]])
+    group = result["SecurityGroups"][0]
 
-    assert (
-        result["SecurityGroups"][0]["IpPermissions"][0]["IpRanges"][0]["Description"]
-        == "testDescription"
-    )
-    assert (
-        result["SecurityGroups"][0]["IpPermissions"][0]["IpRanges"][0]["CidrIp"]
-        == "1.2.3.4/32"
-    )
+    assert group["IpPermissions"][0]["IpRanges"][0]["Description"] == "testDescription"
+    assert group["IpPermissions"][0]["IpRanges"][0]["CidrIp"] == "1.2.3.4/32"
 
     sg = conn.create_security_group(
         GroupName="sg2", Description="Test security group sg1", VpcId=vpc.id
@@ -741,17 +736,10 @@ def test_description_in_ip_permissions():
     )
 
     result = conn.describe_security_groups(GroupIds=[sg["GroupId"]])
+    group = result["SecurityGroups"][0]
 
-    assert (
-        result["SecurityGroups"][0]["IpPermissions"][0]["IpRanges"][0].get(
-            "Description"
-        )
-        is None
-    )
-    assert (
-        result["SecurityGroups"][0]["IpPermissions"][0]["IpRanges"][0]["CidrIp"]
-        == "1.2.3.4/32"
-    )
+    assert group["IpPermissions"][0]["IpRanges"][0].get("Description") is None
+    assert group["IpPermissions"][0]["IpRanges"][0]["CidrIp"] == "1.2.3.4/32"
 
 
 @mock_ec2

--- a/tests/test_ec2/test_security_groups_cloudformation.py
+++ b/tests/test_ec2/test_security_groups_cloudformation.py
@@ -1,0 +1,139 @@
+import boto3
+import sure  # noqa
+from moto import mock_cloudformation, mock_ec2
+
+
+SEC_GROUP_INGRESS = """{
+    "AWSTemplateFormatVersion": "2010-09-09",
+    "Description": "AWS CloudFormation Template to create an EC2 instance",
+    "Parameters": {
+        "VPCId": {
+            "Type": "String",
+            "Description": "The VPC ID",
+            "AllowedPattern": "^vpc-[a-zA-Z0-9]*"
+        }
+    },
+    "Resources": {
+        "SecurityGroup": {
+            "Type": "AWS::EC2::SecurityGroup",
+            "Properties": {
+                "GroupDescription": "Test VPC security group",
+                "GroupName": "My-SG",
+                "VpcId": {
+                    "Ref": "VPCId"
+                }
+            }
+        },
+        "SSHIngressRule": {
+            "Type": "AWS::EC2::SecurityGroupIngress",
+            "Properties": {
+                "CidrIp": "10.0.0.0/8",
+                "Description": "Allow SSH traffic from 10.0.0.0/8",
+                "FromPort": 22,
+                "ToPort": 22,
+                "GroupId": {
+                    "Fn::GetAtt": [
+                        "SecurityGroup",
+                        "GroupId"
+                    ]
+                },
+                "IpProtocol": "tcp"
+            }
+        }
+    }
+}
+"""
+
+
+SEC_GROUP_INGRESS_WITHOUT_DESC = """{
+    "AWSTemplateFormatVersion": "2010-09-09",
+    "Description": "AWS CloudFormation Template to create an EC2 instance",
+    "Parameters": {
+        "VPCId": {
+            "Type": "String",
+            "Description": "The VPC ID",
+            "AllowedPattern": "^vpc-[a-zA-Z0-9]*"
+        }
+    },
+    "Resources": {
+        "SecurityGroup": {
+            "Type": "AWS::EC2::SecurityGroup",
+            "Properties": {
+                "GroupDescription": "Test VPC security group",
+                "GroupName": "My-SG",
+                "VpcId": {
+                    "Ref": "VPCId"
+                }
+            }
+        },
+        "SSHIngressRule": {
+            "Type": "AWS::EC2::SecurityGroupIngress",
+            "Properties": {
+                "CidrIp": "10.0.0.0/8",
+                "FromPort": 22,
+                "ToPort": 22,
+                "GroupId": {
+                    "Fn::GetAtt": [
+                        "SecurityGroup",
+                        "GroupId"
+                    ]
+                },
+                "IpProtocol": "tcp"
+            }
+        }
+    }
+}
+"""
+
+
+@mock_cloudformation
+@mock_ec2
+def test_security_group_ingress():
+    cf_client = boto3.client("cloudformation", region_name="us-east-1")
+    ec2 = boto3.resource("ec2", region_name="us-west-1")
+    ec2_client = boto3.client("ec2", region_name="us-east-1")
+
+    vpc = ec2.create_vpc(CidrBlock="10.0.0.0/16")
+    cf_client.create_stack(
+        StackName="test_stack",
+        TemplateBody=SEC_GROUP_INGRESS,
+        Parameters=[{"ParameterKey": "VPCId", "ParameterValue": vpc.id}],
+        Capabilities=["CAPABILITY_NAMED_IAM"],
+        OnFailure="DELETE",
+    )
+
+    groups = ec2_client.describe_security_groups()["SecurityGroups"]
+    group = [g for g in groups if g["GroupName"] == "My-SG"][0]
+    group["Description"].should.equal("Test VPC security group")
+    len(group["IpPermissions"]).should.be(1)
+    ingress = group["IpPermissions"][0]
+    ingress["FromPort"].should.equal(22)
+    ingress["ToPort"].should.equal(22)
+    ingress["IpProtocol"].should.equal("tcp")
+    ingress["IpRanges"].should.equal(
+        [{"CidrIp": "10.0.0.0/8", "Description": "Allow SSH traffic from 10.0.0.0/8"}]
+    )
+
+
+@mock_cloudformation
+@mock_ec2
+def test_security_group_ingress_without_description():
+    cf_client = boto3.client("cloudformation", region_name="us-east-1")
+    ec2 = boto3.resource("ec2", region_name="us-west-1")
+    ec2_client = boto3.client("ec2", region_name="us-east-1")
+
+    vpc = ec2.create_vpc(CidrBlock="10.0.0.0/16")
+    cf_client.create_stack(
+        StackName="test_stack",
+        TemplateBody=SEC_GROUP_INGRESS_WITHOUT_DESC,
+        Parameters=[{"ParameterKey": "VPCId", "ParameterValue": vpc.id}],
+        Capabilities=["CAPABILITY_NAMED_IAM"],
+        OnFailure="DELETE",
+    )
+
+    groups = ec2_client.describe_security_groups()["SecurityGroups"]
+    group = [g for g in groups if g["GroupName"] == "My-SG"][0]
+    group["Description"].should.equal("Test VPC security group")
+    len(group["IpPermissions"]).should.be(1)
+    ingress = group["IpPermissions"][0]
+    ingress["IpRanges"].should.equal([{"CidrIp": "10.0.0.0/8"}])

